source/browser.civet

```diff
@@ -1,19 +1,28 @@
-// Shimmed node:path and node:module for browser build
+export * from ./main.civet
+import { type CompilerOptions, compile } from ./main.civet
 
-// node:path
-export function dirname(path: string): string
-  path.replace /[^]*\//, ''
-export function resolve(path: string): string
-  path
+// Based on `CoffeeScript.runScripts`, from
+// https://github.com/jashkenas/coffeescript/blob/main/src/browser.coffee
+export function runScripts(type = 'text/civet'): void
+  scripts := window.document.querySelectorAll `script[type=${JSON.stringify type}]`
 
-// node:module
-export function createRequire(path: string): (id: string) => unknown
-  (id: string) =>
-    throw new ReferenceError
-      "Civet comptime does not support 'require' on this platform"
+  for each script, i of scripts
+    options: CompilerOptions :=
+      inlineMap: true
+      js: true
 
-export default {
-  dirname
-  resolve
-  createRequire
-}
+    let code: string
+    if source := (script as HTMLScriptElement).src or script.getAttribute 'data-src'
+      options.filename = source
+      code = fetch source |> await |> .text() |> await
+    else
+      options.filename = script.id or `<script>${i}`
+      code = script.innerHTML
+
+    js := compile code, options |> await
+    window.eval js
+
```

Comment:
We may also want to add an observer for when scripts are appended to the DOM after content loaded and execute them as well.

Reply:
Interesting idea! This might be a performance hit, so perhaps we should only enable this when calling a function, something like `Civet.autoRunScripts(document.body)`? Or does that beat the point of ease-of-use? Related, it seems that [mutation observers](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) need to target a particular node, and we don't necessarily know where `<script>` tags will be added (but `document.head` and `document.body` seem like the most likely places). Perhaps we could make it particularly easy to use, as in `<script src=".../civet/dist/browser.js" data-auto-run-scripts="body">` (where `body` is a query selector).

Reply:
I wouldn't worry too much about the performance and just keep it simple. The direct in browser script is mainly for quick prototyping and should likely be avoided in performance sensitive applications in either case.

Reply:
I agree that `<script type="text/civet">` should only be used for quick prototyping and performance isn't a concern there. My worry is that the browser build might be used in other contexts, too. For example, we use it in our own docs, albeit in a worker thread so none of this DOM code will run. But it seems plausible that someone would want the Civet compiler in browser, without wanting the overhead of mutation observers because they don't use `<script type="text/civet">`. Or do you think the performance impact is small enough? Maybe if we just monitor `document.head`, which I imagine doesn't change often in most applications.

Reply:
OK, I implemented watching for direct children of `head` and `body`, and implemented overrides on the `<script>` tag that includes Civet to turn off all of the functionality of this PR, or selectively just the initial load or the mutation observer. This lets you run scripts when you want, or watch exactly what you want, or turn off everything for performance, if desired.